examples/validator/src/main.rs

```diff
@@ -0,0 +1,78 @@
+//! Run with
+//!
+//! ```not_rust
+//! cargo run -p example-validator
+//!
+//! curl '127.0.0.1:3000?name='
+//! -> Input validation error: [name: Can not be empty]
+//!
+//! curl '127.0.0.1:3000?name=LT'
+//! -> <h1>Hello, LT!</h1>
+//! ```
+
+use axum::{
+    body::{Bytes, Full},
+    extract::Form,
+    handler::get,
+    http::{Response, StatusCode},
+    response::{Html, IntoResponse},
+    Router,
+};
+use serde::Deserialize;
+use std::{convert::Infallible, net::SocketAddr};
+use thiserror::Error;
+use validator::Validate;
+
+#[tokio::main]
+async fn main() {
+    // build our application with a route
+    let app = Router::new().route("/", get(handler));
+
+    // run it
+    let addr = SocketAddr::from(([127, 0, 0, 1], 3000));
+    println!("listening on {}", addr);
+    axum::Server::bind(&addr)
+        .serve(app.into_make_service())
+        .await
+        .unwrap();
+}
+
+#[derive(Debug, Deserialize, Validate)]
+pub struct NameInput {
+    #[validate(length(min = 1, message = "Can not be empty"))]
+    pub name: String,
+}
+
+async fn handler(Form(input): Form<NameInput>) -> Result<Html<String>, ServerError> {
+    input.validate()?;
```

Comment:
I think it would be more idiomatic to create a `ValidatedForm` extractor that wraps `Form` but also calls `.validate()?`. Otherwise users have to remember to check the validation in each handler. Do you think something like that is possible?